File: src/main/java/com/larryhsiao/clotho/ConstSource.java
package com.larryhsiao.clotho;

/**
 * Constant source
 */
public class ConstSource<T> implements Source<T> {

    private final T constant;

    public ConstSource(T constant) {this.constant = constant;}

    @Override
    public T value() throws Exception {
        return constant;
    }
}

File: src/main/java/com/silverhetch/clotho/source/ConstSource.kt
package com.silverhetch.clotho.source

import com.silverhetch.clotho.Source

/**
 * Const source that only keeps the reference.
 * @todo #refactor-1 remove this class which is redundant
 */
class ConstSource<T>(private val value: T) : Source<T> {
    override fun value(): T {
        return value
    }
}